app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java

```diff
@@ -15,31 +18,133 @@
-        modified = false;
         binding = FragmentInfoPetBinding.inflate(inflater, container, false);
+        communication = (InfoPetCommunication) getActivity();
+
         updatePetListeners();
         setCalendarPicker();
         setGenderDropdownMenu();
+        setPetProfileImage();
+        initializePetInfo();
+
+        modified = false;
+
         return binding.getRoot();
     }
 
+    /**
+     * Initialize the pet info.
+     */
+    private void initializePetInfo() {
+        if (pet != null) {
+            Objects.requireNonNull(binding.petName.getEditText()).setText(pet.getName());
+            Objects.requireNonNull(binding.breed.getEditText()).setText(pet.getBreed());
+            Objects.requireNonNull(binding.weight.getEditText()).setText(String.valueOf(pet.getWeight()));
+            Objects.requireNonNull(binding.recommendedKcal.getEditText())
+                .setText(String.valueOf(pet.getRecommendedDailyKiloCalories()));
+            Objects.requireNonNull(binding.washFrequency.getEditText())
+                .setText(String.valueOf(pet.getWashFrequency()));
+            binding.inputBirthMonth.setText(pet.getBirthDate());
+
+            setGender();
+            setPathologies();
+        }
+    }
+
+    /**
+     * Sets the pet pathologies if exists.
+     */
+    private void setPathologies() {
+        if (pet.getPathologies() != null) {
+            Objects.requireNonNull(binding.pathologies.getEditText()).setText(pet.getPathologies());
+        }
+    }
+
+    /**
+     * Sets the gender of the pet.
+     */
+    private void setGender() {
+        if (pet.getGender() == Gender.MALE) {
+            binding.inputGender.setText(R.string.male);
+        } else {
+            binding.inputGender.setText(R.string.female);
+        }
+    }
+
+    /**
+     * Sets the pet profile image.
+     */
+    private void setPetProfileImage() {
+       petProfileImage = binding.imgPet;
+
+       if (petProfileDrawable == null) {
+           assignPetImageToDisplay();
+           isImageModified = false;
+       }
+
+       petProfileImage.setDrawable(petProfileDrawable);
+
+       petProfileImage.setOnClickListener(view -> {
+           communication.makeZoomImage(petProfileImage.getDrawable());
+       });
+    }
+
+    /**
+     * Assigns the pet image to display.
+     */
+    private void assignPetImageToDisplay() {
+        Bitmap petImage = pet.getProfileImage();
+
+        if (petImage == null) {
+            petProfileDrawable = getResources().getDrawable(R.drawable.single_paw);
+        } else {
+            petProfileDrawable = new BitmapDrawable(getResources(), petImage);
+        }
+    }
+
+    /**
+     * Sets the pet profile image drawable.
+     * @param drawable The pet profile drawable to set
+     */
+    public static void setPetProfileDrawable(Drawable drawable) {
+        isImageModified = isImageModified || drawable != petProfileDrawable;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare object references.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)